include/envoy/server/listener_manager.h

```diff
@@ -153,6 +153,19 @@ class ListenerManager {
    */
   virtual bool removeListener(const std::string& name) PURE;
 
+  /*
+   * Stops a listener from accepting new connections without actually removing any of them. This is
```

Comment:
Thanks @ramaraochavali for iterating on this. IMO it's OK to ship with the warming / config change race condition as it's well documented. To that effect, can you actually move some of this documentation into the admin handler docs above under an `.. attention::` block?

Also, @alyssawilk @lizan @PiotrSikora any concerns with shipping with this known race condition until if/when someone complains about it? (Assuming it is well documented.)

/wait

Reply:
move the docs to admin handler docs. PTAL

Reply:
Since the goal of this feature is to help with graceful shutdown, the "stop/drain listeners" operation should be a permanent condition, and allowing LDS to add new listeners would render it a bit useless, IMHO.

I think that making this a permanent condition (vs racy) would require adding a boolean to listener manager, which would result in all new LDS resources in a given direction being rejected, and possibly extra check to drain new listeners, immediately after they're warmed, in case they were warming at the time of the request (although, looking at `ListenerManagerImpl::removeListener`, it seems that it should be possible to destroy warming up listeners before they warm up).

Reply:
Another option I can think of if we want to stop LDS to add/remove  is to maintain a list of stopped listeners (not just direction because we may want to stop listener by name in future) and block add/remove and handle in warm up callback if the listener is stopped. I have been experimenting with it. Let me push that change and see if that is better.

Reply:
@mattklein123  pushed that change. Please see if this is better or any other condition is missed?